### PR TITLE
quantum-espresso: enable linking with armpl-gcc and acfl for BLAS and FFT

### PR DESCRIPTION
**Pull request by annop-w**

### CODE REVIEW

File: var/spack/repos/builtin/packages/quantum-espresso/package.py [hunk 1]
```diff
@@ -407,6 +409,18 @@ def cmake_args(self):
...
+        # https://github.com/QEF/q-e/commit/cf74c944fe8498dd1c423902267cea90ba5d036f
+        if "^armpl-gcc" in spec:
+            cmake_args.append(self.define("BLA_VENDOR", "All"))
+            cmake_args.append(self.define("BLAS_LIBRARIES", spec["armpl-gcc"].libs.joined(";")))
+            cmake_args.append(self.define("LAPACK_LIBRARIES", spec["armpl-gcc"].libs.joined(";")))
+        elif "^acfl" in spec:
+            cmake_args.append(self.define("BLA_VENDOR", "All"))
+            cmake_args.append(self.define("BLAS_LIBRARIES", spec["acfl"].libs.joined(";")))
+            cmake_args.append(self.define("LAPACK_LIBRARIES", spec["acfl"].libs.joined(";")))
+
```
Comment (ye-luo) [inline var/spack/repos/builtin/packages/quantum-espresso/package.py:418]: Could you describe the issue when addition cmake_args are not added?

Comment (annop-w) [reply var/spack/repos/builtin/packages/quantum-espresso/package.py:418]: From what I tested, without adding the pointers to BLAS and LAPACK libraries cmake script is unable to find ArmPL libraries (provided by armpl-gcc or acfl package). The issue with having to set BLA_VENDOR to All is the following: after cmake detects ArmPL libraries it seems to set BLA_VENDOR to Arm. That prevents the cmake script which searches for vanilla scalapack (BLA_VENDOR must be set to either Generic or All).

Comment (ye-luo) [reply var/spack/repos/builtin/packages/quantum-espresso/package.py:418]: Based on the comment, this issue has been addressed in QE develop and thus the added workaround is only needed for version up-to 7.1. If my understanding is correct, please add a version guard.

Comment (annop-w) [reply var/spack/repos/builtin/packages/quantum-espresso/package.py:418]: Version guard to set BLA_VENDOR added. I also merged the two if-blocks together.

**Review by bellenlau — COMMENTED** [review]
Required changes have been implemented.

**Review by bellenlau — DISMISSED** [review]

**Comment by annop-w** [comment]
@tldahlgren Could you please help taking a look at this ? One of the job failed, but I don't think it's related to this patch. Thank you.

**Comment by tldahlgren** [comment]
> @tldahlgren Could you please help taking a look at this ? One of the job failed, but I don't think it's related to this patch. Thank you.

Looks like it's all working now.

**Comment by tldahlgren** [comment]
@bellenlau Is this PR ready to be merged now?

**Review by bellenlau — APPROVED** [review]

**Review by ye-luo — APPROVED** [review]

**Review by alalazo — APPROVED** [review]
Approved by maintainers. Code added LGTM.

**Pull request merged**